ReSwift/Utils/Synchronized.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  Synchronized.swift
+//  ReSwift
+//
+//  Created by Basem Emara on 2020-08-18.
+//  https://basememara.com/creating-thread-safe-generic-values-in-swift/
+//
+//  Copyright © 2020 ReSwift Community. All rights reserved.
+//
+
+import Foundation
+
+/// An object that manages the execution of tasks atomically.
+struct Synchronized<Value> {
+    private let mutex = DispatchQueue(label: "reswift.github.io.ReSwift.Utils.Synchronized", attributes: .concurrent)
+    private var _value: Value
+    init(_ value: Value) {
+        self._value = value
+    }
+    /// Returns or modify the thread-safe value.
+    var value: Value { mutex.sync { _value } }
```

Comment:
For the request by @dani-mp maybe this will work. We lose the ability to throw in the setter but we have no need for that in this use case.
```suggestion
    var value: Value {
        get { mutex.sync { _value } }
        set { mutex.sync(flags: .barrier) { _value = newValue }
    }
```

Reply:
Unfortunately, there was something it didn't like about that. The unit test failed for the concurrency with `Thread 9: EXC_BAD_ACCESS (code=EXC_I386_GPFLT)`. I think the `+=` expression tricks it and no longer part of the closure in the setter. This is what stopped me from creating it as a property wrapper.

Reply:
Interesting. Some sort of optimization compiler optimization happening. Might be interesting to take a look at some assembled output of a minimal test case to see what its doing.